packages/react-api/src/hooks/tileset/useSpatialFilterTileset.js

```diff
@@ -0,0 +1,137 @@
+import { _applySpatialFilterToTileContent, debounce } from '@carto/react-core/';
+import { selectSpatialFilter } from '@carto/react-redux';
+import { useCallback, useRef, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { getTileId } from '../utils';
+
+const EMPTY_OBJ = {};
+
+const createEmptyAnalysedFeatures = () => ({
+  points: new Map(),
+  polygons: new Map(),
+  lines: new Map()
+});
+
+export default function useSpatialFilterTileset({ source, tilesCacheRef }) {
+  // Stores already analysed features. Used to avoid inconsistencies between tiles.
+  const analysedFeaturesRef = useRef(createEmptyAnalysedFeatures());
+
+  // For forcing tile filtering
+  const [reloadCounter, setReloadCounter] = useState(0);
+
+  const debounceIdReloadRef = useRef(null);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedReload = useCallback(
+    debounce((shouldClearTilesCache) => {
+      if (shouldClearTilesCache) {
+        tilesCacheRef.current = {};
+      }
+      setReloadCounter((oldState) => (oldState ? 0 : 1));
+    }, 25),
+    []
+  );
+
+  // Stores getFilterValue typed array for each tileId
+  const [spatialFilterBuffers, setSpatialFilterBuffers] = useState(EMPTY_OBJ);
+
+  // Check incrementalDebounce definition comments
+  const debounceIdSetSpatialFilterBuffersRef = useRef(null);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedSetSpatialFilterBuffers = useCallback(
+    incrementalDebounce(setSpatialFilterBuffers, 50),
+    []
+  );
+
+  const spatialFilterGeometry = useSelector((state) =>
+    selectSpatialFilter(state, source?.id)
+  );
+
+  const currentFilterGeometryRef = useRef(spatialFilterGeometry);
+
+  let hasGeometryChanged = currentFilterGeometryRef.current !== spatialFilterGeometry;
+  // Reset everything if geom has changed
+  // Do not use useEffect because it's slow for this case
+  if (hasGeometryChanged && spatialFilterGeometry) {
+    currentFilterGeometryRef.current = spatialFilterGeometry;
+    if (debounceIdReloadRef.current) {
+      clearTimeout(debounceIdReloadRef.current);
+      debounceIdReloadRef.current = null;
+    }
+    if (debounceIdSetSpatialFilterBuffersRef.current) {
+      clearTimeout(debounceIdSetSpatialFilterBuffersRef.current);
+      debounceIdSetSpatialFilterBuffersRef.current = null;
+    }
+    tilesCacheRef.current = {};
+    analysedFeaturesRef.current = createEmptyAnalysedFeatures();
+  }
+
+  const applySpatialFilter = useCallback(
+    (props) => {
+      // props.data is the same as props.tile.content
+      let { data, uniqueIdProperty = 'cartodb_id' } = props;
```

Comment:
instance of this `uniqueIdProperty = 'cartodb_id'` I think that we could use the functions that returns que unique property id